las_geoh5/import_las.py

```diff
@@ -0,0 +1,219 @@
+#
+#  All rights reserved.
+#
+from __future__ import annotations
+
+import warnings
+from pathlib import Path
+
+import lasio
+import numpy as np
+from geoh5py import Workspace
+from geoh5py.groups import DrillholeGroup
+from geoh5py.objects import Drillhole
+from geoh5py.shared.concatenation import ConcatenatedDrillhole
+
+
+def get_depths(lasfile: lasio.LASFile):
+    """
+    Get depth data from las file.
+
+    :param lasfile: Las file object.
+
+    :return: Depth data.
+    """
+
+    try:
+        return lasfile["DEPTH"]
+    except KeyError:
+        pass
+
+    try:
+        return lasfile["DEPT"]
+    except KeyError as err:
+        raise KeyError(
+            "In order to import data to geoh5py format, .las files "
+            "must contain a depth curve named 'DEPTH' or 'DEPT'."
+        ) from err
+
+
+def find_copy_name(workspace: Workspace, basename: str, start: int = 1):
+    """
+    Augment name with increasing integer value until no entities found.
+
+    :param workspace: A geoh5py.Workspace object.
+    :param basename: Existing name of entity in workspace.
+    :param start: Integer name augmenter to test for existence.
+
+    :returns: Augmented name of the earliest non-existent copy in workspace.
+    """
+
+    name = f"{basename} ({start})"
+    obj = workspace.get_entity(name)
+    if obj and obj[0] is not None:
+        find_copy_name(workspace, basename, start=start + 1)
+    return name
+
+
+def add_survey(survey: str | Path, drillhole: Drillhole | ConcatenatedDrillhole):
+    """
+    Import survey data from csv or las format and add to drillhole.
+
+    :param survey: Path to a survey file stored as .csv or .las format.
+    :param drillhole: Drillhole object to append data to.
+
+    :return: Updated drillhole object.
+    """
+
+    if isinstance(survey, str):
+        survey = Path(survey)
+
+    if survey.suffix == ".las":
+        file = lasio.read(survey, mnemonic_case="preserve")
+        try:
+            surveys = np.c_[get_depths(file), file["DIP"], file["AZIM"]]
+            if len(drillhole.surveys) == 1:  # type: ignore
+                drillhole.surveys = surveys  # type: ignore
+        except KeyError:
+            warnings.warn(
+                "Attempted survey import failed because data read from "
+                ".las file did not contain the expected 3 curves 'DEPTH'"
+                ", 'DIP', 'AZIM'."
+            )
+    else:
+        surveys = np.genfromtxt(survey, delimiter=",", skip_header=0)
+        if surveys.shape[1] == 3:  # type: ignore
+            drillhole.surveys = surveys  # type: ignore
+        else:
+            warnings.warn(
+                "Attempted survey import failed because data read from "
+                "comma separated file did not contain the expected 3 "
+                "columns of depth/dip/azimuth."
+            )
+
+    return drillhole
+
+
+def create_or_append_drillhole(  # pylint: disable=too-many-locals

```

Comment:
**thought:** too-many-locals is often an indication that the funtion needs to be broken down into smaller functions

Reply:
done.